src/js/utils/MesosStateUtil.js

```diff
@@ -26,6 +26,19 @@ const MesosStateUtil = {
   },
 
   /**
+   * @param {{frameworks:array,completed_frameworks:array}} state
+   * @param {string} frameworkID
+   * @returns {{executors:Array, completed_executors:Array}|null} framework
+   */
+  getFramework(state, frameworkID) {
+    const {frameworks, completed_frameworks} = state;
+    return [].concat(frameworks, completed_frameworks).find(
+      function (framework) {
+        return framework != null && framework.id === frameworkID;
```

Comment:
and it looks like we don't need that check anymore :) Unless `state.frameworks` or `state.completed_frameworks` give null, in case of `null` defaults won't work 😢